test/integration/ThreeChainEndToEnd.t.sol

```diff
@@ -53,64 +63,40 @@ contract ThreeChainEndToEndDeployment is EndToEndUtils {
         vm.label(address(adapterCToA), "AdapterCToA");
     }
 
-    /// @param sameChain Whether the chain B equals chain C, such that sB points to Centrifuge ID C == 7
-    function _setSpokeB(bool sameChain) internal {
-        if (sameChain) {
-            _setSpoke(deployC, CENTRIFUGE_ID_C, sB);
-        } else {
+    function _setSpokes(CrossChainDirection direction) internal {
+        if (direction == CrossChainDirection.WithIntermediaryHub) {
             _setSpoke(deployB, CENTRIFUGE_ID_B, sB);
+            _setSpoke(deployC, CENTRIFUGE_ID_C, sC);
+        } else if (direction == CrossChainDirection.FromHub) {
+            _setSpoke(deployB, CENTRIFUGE_ID_B, sB);
+            _setSpoke(deployA, CENTRIFUGE_ID_A, sC);
+        } else if (direction == CrossChainDirection.ToHub) {
+            _setSpoke(deployA, CENTRIFUGE_ID_A, sB);
+            _setSpoke(deployC, CENTRIFUGE_ID_C, sC);
         }
     }
-}
-
-/// @title  Three Chain End-to-End Use Cases
-/// @notice Test cases for the three-chain setup
-contract ThreeChainEndToEndUseCases is ThreeChainEndToEndDeployment {
-    using CastLib for *;
-    using MessageLib for *;
 
     /// @notice Configure the third chain (C) with assets
-    /// @param sameChain Whether the chain B equals chain C, such that sB points to Centrifuge ID C == 7
-    /// forge-config: default.isolate = true
-    function testConfigureAssets(bool sameChain) public {
-        _setSpokeB(sameChain);
+    function _testConfigureAssets(CrossChainDirection direction) internal {
+        _setSpokes(direction);
 
         _configureAsset(sB);
-
-        if (!sameChain) {
-            _configureAsset(sC);
-        }
+        _configureAsset(sC);
     }
 
     /// @notice Configure a pool with support for all three chains
-    /// forge-config: default.isolate = true
-    function testConfigurePool(bool sameChain) public {
-        _setSpokeB(sameChain);
+    function _testConfigurePool(CrossChainDirection direction) internal {
+        _setSpokes(direction);
 
         _configurePool(sB);
-
-        if (!sameChain) {
-            _configurePool(sC);
-        }
-    }
-
-    /// @notice Test transferring shares between Chain B and Chain C via Hub A
-    /// forge-config: default.isolate = true
-    function testCrossChainTransferShares() public {
-        _testCrossChainTransferShares(false);
+        _configurePool(sC);
     }
 
     /// @notice Test transferring shares between Chain B and Chain C via Hub A
-    /// forge-config: default.isolate = true
-    function testCrossChainTransferSharesSameChain() public {
-        _testCrossChainTransferShares(true);
-    }
-
-    /// @notice Test transferring shares between Chain B and Chain C via Hub A
-    function _testCrossChainTransferShares(bool sameChain) internal {
+    function _testCrossChainTransferShares(CrossChainDirection direction) internal {
```

Comment:
Is the fuzzer not working with enums? Is it related to the Boolean issue you found?

Reply:
Ah so it performs a lot of duplications, right?

Wonder why we have specific tests for each variant instead of fuzzing the enum

Reply:
> Ah so it performs a lot of duplications, right?

Yeah it runs for the configured fuzz_runs even if it exhausted all possible values.

> Wonder why we have specific tests for each variant instead of fuzzing the enum

This way, each test will run once and debugging is much smoother!